packages/mui-base/src/Select/Select.spec.tsx

```diff
@@ -35,49 +29,25 @@ function CustomRoot<OptionValue extends {}, Multiple extends boolean>(
   return <div {...other} />;
 }
 
-function CustomPopper<OptionValue extends {}, Multiple extends boolean>(
-  props: WithOptionalOwnerState<SelectPopperSlotProps<OptionValue, Multiple>>,
+function CustomPopup<OptionValue extends {}, Multiple extends boolean>(
+  props: SelectPopupSlotProps<OptionValue, Multiple>,
 ) {
   const { ownerState, ...other } = props;
-  return <Popper {...other} />;
+  return <div {...other} />;
 }
 
 const SelectRootComponentOverridesTest = (
   <Select
     slots={{
       root: CustomRoot,
-      listbox: 'ul',
-      popper: Popper,
-    }}
-  />
-);
-
-const SelectPopperComponentOverridesTest = (
-  <Select
-    slots={{
-      popper: CustomPopper,
-    }}
-  />
-);
-
-function InvalidPopper({ requiredProp }: { requiredProp: string }) {
-  return <div />;
-}
```

Comment:
Isn't this one still relevant?

Reply:
It used to be required to pass in a Popper or a component with a similar API to the `slots.popper`. Now it can be any element, as whatever is passed in, will become the root slot of the Popup.
So after the change, the `popup` slot is typed as any other slot (`React.ElementType`), and we don't typecheck any props there.